clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -0,0 +1,1064 @@
+                log.trace("{}: KafkaClient#poll retrieved {} response(s)", clientId, responses.size());
+
+                // Update the current time and handle the latest responses.
+                now = time.milliseconds();
+                handleResponses(now, responses, callsInFlight, correlationIdToCalls);
+            }
+            int numTimedOut = 0;
+            synchronized (this) {
+                numTimedOut += timeoutCalls(Long.MAX_VALUE, newCalls);
+            }
+            numTimedOut += timeoutCalls(Long.MAX_VALUE, correlationIdToCalls.values());
+            if (numTimedOut > 0) {
+                log.debug("{}: timed out {} remaining operations.", clientId, numTimedOut);
+            }
+            closeQuietly(client, "KafkaClient");
+            closeQuietly(metrics, "Metrics");
+            log.debug("{}: exiting AdminClientRunnable thread.", clientId);
+        }
+
+        void call(Call call, long now) {
+            if (log.isDebugEnabled()) {
+                log.debug("{}: queueing {} with a timeout {} ms from now.",
+                    clientId, call, call.deadlineMs - now);
+            }
+            synchronized (this) {
+                newCalls.add(call);
+            }
+            client.wakeup();
+        }
+    }
+
+    @Override
+    public CreateTopicResults createTopics(final Collection<NewTopic> newTopics,
+                                           final CreateTopicsOptions options) {
+        final Map<String, KafkaFutureImpl<Void>> topicFutures = new HashMap<>(newTopics.size());
+        for (NewTopic newTopic : newTopics) {
+            topicFutures.put(newTopic.name(), new KafkaFutureImpl<Void>());
+        }
+        final Map<String, CreateTopicsRequest.TopicDetails> topicsMap = new HashMap<>(newTopics.size());
+        for (NewTopic newTopic : newTopics) {
+            topicsMap.put(newTopic.name(), newTopic.convertToTopicDetails());
+        }
+        final long now = time.milliseconds();
+        runnable.call(new Call("createTopics", calcDeadlineMs(now, options.timeoutMs()),
+            new ControllerNodeProvider()) {
+
+            @Override
+            public AbstractRequest.Builder createRequest(int timeoutMs) {
+                return new CreateTopicsRequest.Builder(topicsMap, timeoutMs, options.validateOnly());
+            }
+
+            @Override
+            public void handleResponse(AbstractResponse abstractResponse) {
+                CreateTopicsResponse response = (CreateTopicsResponse) abstractResponse;
+                // Handle server responses for particular topics.
+                for (Map.Entry<String, CreateTopicsResponse.Error> entry : response.errors().entrySet()) {
+                    KafkaFutureImpl<Void> future = topicFutures.get(entry.getKey());
+                    if (future == null) {
+                        log.warn("Server response mentioned unknown topic {}", entry.getKey());
```

Comment:
It's kind of awkward because other futures might have been completed successfully at this point.  Assuming that there is a bug where the server is responding with an incorrect topic name, this will be caught by means of the sanity check at the bottom that fails futures which haven't been completed by the response.